### PR TITLE
ISAICP-3780: general content page - event date facet

### PR DESCRIPTION
**Pull request by brummbar**

### CODE REVIEW

File: web/modules/custom/joinup_event/joinup_event.behat.inc [hunk 1]
```diff
@@ -31,4 +32,36 @@ class EventSubContext extends DrupalSubContextBase implements DrupalSubContextIn
...
+  public static function massageEventFieldsBeforeNodeCreate(BeforeNodeCreateScope $scope) {
+    $node = $scope->getEntity();
+
+    if ($node->type !== 'event') {
+      return;
+    }
+
+    foreach (['start date', 'end date'] as $field) {
+      if (isset($node->$field)) {
+        $date = strtotime($node->$field);
```
Comment (pfrenssen) [inline web/modules/custom/joinup_event/joinup_event.behat.inc]: Great solution! Let the PHP standard library do the heavy lifting!

File: composer.json [hunk 2]
```diff
@@ -150,11 +150,13 @@
                 "Don't compute children and parents of a new term @see https://www.drupal.org/node/2845361": "https://www.drupal.org/files/issues/2845361-2.patch",
                 "Incorrect field name is used in views integration for multi-value base fields @see https://www.drupal.org/node/2846614": "https://www.drupal.org/files/issues/2846614-112.patch",
                 "New event dispatch: a migrated entity is about to be saved @see https://www.drupal.org/node/2890012": "https://www.drupal.org/files/issues/2890012-8.patch",
-                "Remove cached languages from ContentEntityBase @see https://www.drupal.org/node/2303877": "https://www.drupal.org/files/issues/2303877-22.patch"
+                "Remove cached languages from ContentEntityBase @see https://www.drupal.org/node/2873480": "https://www.drupal.org/files/issues/2303877-22.patch"
```
[the remark this answers is not in the document]
Comment (brummbar) [reply composer.json]: Ow my bad!

File: tests/features/content_overview.feature [hunk 3]
```diff
@@ -162,3 +162,120 @@ Feature: Content Overview
...
+      | Frozen Barbershop    |
+      | Flying Official Fish |
+      | Bitter Finger        |
+    And the "Event date" inline facet should allow selecting the following values "Upcoming events (2), My events (1), All events"
+
+    When I am an anonymous user
+    And I am on the homepage
+    And I click "Events, discussions, news ..."
+    And I click "Event"
+    Then the "Event date" inline facet should allow selecting the following values "Past events (3), Upcoming events (2)"
```
[the remark this answers is not in the document]
Comment (brummbar) [reply tests/features/content_overview.feature]: The step tests that only the specified facet items are visible, so it asserts that "My events" is not there.

**Comment by pfrenssen** [comment]
My remarks will be handled in a followup: https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-4133

**Pull request merged**